main.go

```diff
@@ -0,0 +1,382 @@
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type config []struct {
+	Targets []string `yaml:"targets"`
+}
+
+func main() {
+	err := validateEnvironmentVariables()
+	if err != nil {
+		log.WithError(err).Error("Environment variable validation failed")
+		err = sendMattermostErrorNotification(err, "Environment variable validation failed")
+		if err != nil {
+			log.WithError(err).Error("Failed to send Mattermost error notification")
+		}
+		return
+	}
+
+	err = thanosStoreDiscovery()
+	if err != nil {
+		log.WithError(err).Error("Failed to run Thanos store discovery")
+		err = sendMattermostErrorNotification(err, "The Thanos store discovery failed")
+		if err != nil {
+			log.WithError(err).Error("Failed to send Mattermost error notification")
+		}
+		return
+	}
+}
+
+// validateEnvironmentVariables is used to validate the environment variables needed by Thanos store discovery.
+func validateEnvironmentVariables() error {
+	privateHostedZoneID := os.Getenv("PRIVATE_HOSTED_ZONE_ID")
+	if len(privateHostedZoneID) == 0 {
+		return errors.Errorf("PRIVATE_HOSTED_ZONE_ID environment variable is not set")
+	}
+
+	thanosNamespace := os.Getenv("THANOS_NAMESPACE")
+	if len(thanosNamespace) == 0 {
+		return errors.Errorf("THANOS_NAMESPACE environment variable is not set")
+	}
+
+	thanosDeploymentName := os.Getenv("THANOS_DEPLOYMENT_NAME")
+	if len(thanosDeploymentName) == 0 {
+		return errors.Errorf("THANOS_DEPLOYMENT_NAME environment variable is not set.")
+	}
+
+	thanosConfigMapName := os.Getenv("THANOS_CONFIGMAP_NAME")
+	if len(thanosConfigMapName) == 0 {
+		return errors.Errorf("THANOS_CONFIGMAP_NAME environment variable is not set.")
+	}
+
+	mattermostAlertsHook := os.Getenv("MATTERMOST_ALERTS_HOOK")
+	if len(mattermostAlertsHook) == 0 {
+		return errors.Errorf("MATTERMOST_ALERTS_HOOK environment variable is not set.")
+	}
+	return nil
+}
+
+// thanosStoreDiscovery is used to keep Thanos up to date with deployed Thanos Query targets.
+func thanosStoreDiscovery() error {
+	privateHostedZoneID := os.Getenv("PRIVATE_HOSTED_ZONE_ID")
+	thanosNamespace := os.Getenv("THANOS_NAMESPACE")
+	thanosDeploymentName := os.Getenv("THANOS_DEPLOYMENT_NAME")
+	thanosConfigMapName := os.Getenv("THANOS_CONFIGMAP_NAME")
+
+	log.Infof("Getting Route53 records for hostedzone %s", privateHostedZoneID)
+	records, err := listAllRecordSets(privateHostedZoneID)
+	if err != nil {
+		return errors.Wrap(err, "Unable to get the existing Thanos Route53 records")
+	}
+
+	thanosRecords := getThanosTargets(records)
+	if len(thanosRecords) < 1 {
+		log.Info("No Thanos records to register, canceling run")
+		return nil
+	}
+	route53Targets := addPortToTargets(thanosRecords)
+
+	// This part can be used for local testing
+	// kubeconfig := filepath.Join(
+	// 	os.Getenv("HOME"), ".kube", "config",
+	// )
+
+	// config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	// if err != nil {
+	// 	return err
+	// }
```

Comment:
Can you remove the commented code or instead add a dev flag so that it is only used when run with a certain flag? Leaving commented code around is generally a bad practice

Reply:
Thanks for that, makes total sense. I removed the commented code, created a separate function to get the clientset and added an environment variable that can be set for dev local testing.